src/HttpClient.ts

```diff
@@ -70,6 +72,20 @@ export class HttpClient {
         }
     }
 
+    public static getHttpToHttpsProxyConfig(
+        proxyConfig: IProxyConfig | false | undefined
+    ): HttpsProxyAgent | undefined {
+        if (
+            !proxyConfig ||
+            !proxyConfig.host ||
+            !proxyConfig.port ||
+            (proxyConfig.protocol && proxyConfig.protocol.includes('https'))
```

Comment:
This check repeats here and in getAxiosProxyConfig

Reply:
not exactly, here we check if the protocol **is** 'https' and return undefined if they do (i.e use default)
at getAxiosProxyConfig we check if the protocol **is not** 'https', i.e getHttpToHttpsProxyConfig will not return undefined, therefore we need to return false in order to disable the default proxy handling